include/circt/Dialect/Calyx/CalyxStructure.td

```diff
@@ -10,7 +10,29 @@
 //
 //===----------------------------------------------------------------------===//
 
+def ProgramOp : CalyxContainer<"program", [
+    IsolatedFromAbove,
+    SymbolTable
+  ]> {
+  let summary = "Calyx Program";
+  let description = [{
+    The "calyx.program" operation represents an overall Calyx program,
+    containing a list of Calyx components. This must include a "main"
+    component, the entry point of the program.
+  }];
+  let arguments = (ins);
+
+  let extraClassDeclaration = [{
+    /// Returns the main component, representing the
+    /// entry point of the Calyx program.
+    Operation *getMainComponent() { return lookupSymbol("main"); }
```

Comment:
I think there is a templated version like `lookupSymbol<ComponentOp>`, would that work here so you get back a ComponentOp instead of an Operation *?